### PR TITLE
WIP: use refreshVersions inside Splitties

### PR DESCRIPTION
**Pull request by jmfayard**
Status: blocked by https://github.com/ben-manes/gradle-versions-plugin/issues/334

---

To check dependencies versions, run

  `gw --scan compileDebugSources`

Results before the change
https://gradle.com/s/jhmwamwn5pjag/dependencies?expandAll

Results after the change
https://gradle.com/s/bds2xao6fbmqa/dependencies?expandAll

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -15,3 +19,27 @@ allprojects {
...
+
+/**
+ * Use ./gradlew refreshVersions to find available updates
+ * See https://github.com/jmfayard/buildSrcVersions/issues/77
+ */
+buildSrcVersions {
+    // Configuration: https://github.com/jmfayard/buildSrcVersions/issues/53
+    rejectVersionIf {
+        isNonStable(candidate.version) && isNonStable(currentVersion).not()
+    }
```
[the remark this answers is not in the document]
Comment (jmfayard) [reply build.gradle.kts]: Indeed that makes a lot of sense

File: gradle.properties [hunk 2]
```diff
@@ -28,3 +28,128 @@ kotlin.native.ignoreDisabledTargets=true
...
+version.kotlinx-coroutines-test=1.3.1
+#                   # available=1.3.2
+version.androidx.test.espresso=3.1.1
+version.com.squareup.retrofit2=2.5.0
+version.play-services-location=16.0.0
+version.play-services-wearable=16.0.1
+version.androidx.media..media=1.0.0
+version.androidx.print..print=1.0.0
+version.androidx.recyclerview=1.0.0
+version.localbroadcastmanager=1.0.0
```
[the remark this answers is not in the document]
Comment (jmfayard) [reply gradle.properties]: The way to be explicit with my plugin is to use `useFdqnFor(...)`.
I did it for a few dependencies, tell me what you think

**Comment by jmfayard** [comment]
I pushed a new version. Next steps would be:

- [ ] Release a version with `isStable()`
- [ ] Complete the `useFqdnFor(...)` solution if you are happy with it
- [ ] Inline all versions in `Dependencies.kt` since they are overwritten anyway

File: build.gradle.kts [hunk 3]
```diff
@@ -3,13 +3,20 @@
...
+
 plugins {
     id("de.fayard.buildSrcVersions").version("0.6.3")
     id ("com.gradle.build-scan").version("2.4.2") 
 }
 
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
+// Libs.kotlinVersion comes from gradle.properties. Keep in sync with buildSrc/build.gradle.kts
+Libs.kotlinVersion = findProperty("version.kotlin-stdlib-common") as String
```
Comment (LouisCAD) [inline build.gradle.kts]: Why not just use `version.kotlin`?

File: build.gradle.kts [hunk 4]
```diff
@@ -15,3 +26,36 @@ allprojects {
...
+    // Configuration: https://github.com/jmfayard/buildSrcVersions/issues/53
+    rejectVersionIf {
+        isStable(currentVersion) && isNonStable(candidate.version)
+    }
+    useFqdnFor(*Libs.AndroidX.useFdqnFor)
+}
+
+
+/** 
+For investing build issue and bug reports, run
```
Comment (LouisCAD) [inline build.gradle.kts]:
```suggestion
For investigating build issue and bug reports, run 
```

File: gradle.properties (hunk 2 above)
[the remark this answers is not in the document]
Comment (LouisCAD) [reply gradle.properties]: `useFdqnFor(…)` is great! I can continue the work for other AndroidX artifacts.

File: gradle.properties [hunk 5]
```diff
@@ -38,112 +43,72 @@ splitties.version=3.0.0-dev-038
 plugin.com.github.ben-manes.versions=0.25.0
 plugin.org.gradle.kotlin.kotlin-dsl=1.2.9
 #                       # available=1.3.1
-plugin.de.fayard.buildSrcVersions=0.6.3
+plugin.de.fayard.buildSrcVersions=0.6.4
```
Comment (LouisCAD) [inline gradle.properties]: Isn't that a duplicate version number with the one defined in root project's `build.gradle.kts`?

File: build.gradle.kts (hunk 3 above)
Comment (jmfayard) [reply build.gradle.kts]: because it's not generated by `:refreshVersions` and would be overwritten next time we run the task

**Comment by jmfayard** [comment]
v0.6.4 is now released!

**Pull request closed without merging**